File: Assembler/Java/symbolsandvariables.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class symbolsandvariables{
    public static void main(String[] args) throws IOException       
    {
        try{                                                                                   // The IOException is a checked exception that can be thrown by the main method if there is an error reading or writing to an input or output stream
            File read=new File("nowhitespaces.asm");                                  // creating a new instance read to read nowhitespaces.asm 
            Scanner in=new Scanner(read);  
            List<String> variables= new ArrayList<String>();                                                    // creating a new instance in to read from file nowhitespaces.asm
            FileWriter writer=new FileWriter("symbolandvariables.asm");               // creating a new instance writer to write in symbolandvariables.asm
            while (in.hasNextLine()) {                                                         // if there is EOF returns false otherwise
                String data = in.nextLine();                                                   // reading each line
                
                if(data.charAt(0)=='@')
                {
                    int f= (int)(data.charAt(1));                                        // as variable name starts with a char so checking whether first char is a char or not
                    if(((f>64)&&(f<91))||((f>96)&&(f<123)))
                    {
                        if(!variables.contains(data.substring(1))){
                            variables.add(data.substring(1));
                            writer.write(data.substring(1));                            // write to the file from index 1 removing @                                   
                            writer.write('\n');
                            System.out.println(data);
                        }
                    }
                }
              }
            writer.close();
            in.close();
        } catch(FileNotFoundException e){                                                      // if file nowhitespaces.asm is not found
            System.out.println("An error occured.");
            e.printStackTrace();
        }
    }
}

File: Assembler/C/symbols and variables.c
#include<stdio.h>
#include<string.h>
int main()
{
	char b[100],c[100];
	FILE *fp,*file; //filepointer
	fp=fopen("whitespace.asm","r");
	file=fopen("symbolsandvariables.asm","w");
	while( fgets ( b, 100, fp ) != NULL )//CAN ALSO USE EOF(END OF FILE)
     {
		 if(b[0]=='(')
		 {
			 fputs(b,file);
		 }
		 else if(b[0]=='@')
		 {
			int ascii=b[1];
			if(((ascii>64)&&(ascii<91))||((ascii>96)&&(ascii<123)))
			{
				for(int i=1;i<strlen(b);i++)
				{
					c[i-1]=b[i];
					c[i+1]='\0';
				}
				fputs(c,file);
			}
		 }
	 }
	 fclose(fp);
     fclose(file);
	 printf("Data successfully written in file symbolandvariables.asm\n");
     printf("The file is now closed.") ;
     return 0;
}
